reactioncommerce/reaction-component-library
# Update theme to have nested objects and document all component themes

Impact: **minor**  
Type: **feature**

This is part 2 of #315

## Changes
- All component theme variables are now grouped within `rui_components.${ComponentName}` in the default theme (and expected to be in this pattern in any custom theme)
- Added a "Theming Components" section in the style guide
- Added a "Theme" section with "Typography" subsection to the bottom of every component markdown. These have a table of all theme variables the component uses, what their default values are, and what they do.
- Minor cleanup of theme variables and their values in a few places

## Breaking changes
This is breaking in the sense that the structure of the theme object has changed.

## Testing
- Verify that all components still look the same when using the default theme.
- Skim the "Theme" section in the docs for each component
- Click on each component in the style guide and make sure there are no errors in the console
- Review the general "Theming Components" article
